### PR TITLE
Avoid getting the theme if we have it already

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | replaces #11755                       |
| License         | MIT                                                          |

Just finishing it up.

Kudos to @jacquesbh for catching and fixing this huge waste of resources :beers:

### CODE REVIEW

**Comment by jacquesbh** [comment]
Nice! Finally!

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php [hunk 1]
```diff
@@ -22,26 +22,31 @@
 
 final class ChannelBasedThemeContext implements ThemeContextInterface
 {
+    private null|false|ThemeInterface $theme = false;
```
Comment (coldic3) [inline src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php:25]: A separate property for boolean would be better IMO eg. `$alreadyCalled`.

Comment (jacquesbh) [reply src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php:25]: I disagree. `false` means "no theme", `null` means not set, and the `ThemeInterface` means we have one.

All 3 values are logical.

Comment (coldic3) [reply src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php:25]: Hmmm I see the difference between the 3 of them but `no theme` also means nothing has been set yet, and `not set` means there is `no theme` so it's tricky at first glance – at least for me :smile:

Comment (mbabker) [reply src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php:25]: Defaulting it to false is the closest thing PHP has to an `is_initialized($this->prop);` type of check.  If it's false, the property hasn't been initialized yet and whatever code needs to seed it needs to run.  Once it's null or an object, it's initialized.

Comment (jakubtobiasz) [reply src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php:25]: I'm ok with the current implementation if anybody asks me 😂

**Review by Rafikooo — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jan and Jacques! :1st_place_medal:

**Pull request merged**